### PR TITLE
Fix/1154  check empty htlml tags in comments

### PR DESCRIPTION
**Pull request by basilelegal**
check empty htlml tags in comments

Ref #1154

### CODE REVIEW

**Comment by basilelegal** [comment]
> Can you add tests where:
> 
>     * `raw_content` contains bad html, without text, example `<p></i>` (empty comment)
> 
>     * `raw_content` contains bad html, with test, example `<p>hello</i>` (there is comment)
> 
>     * `raw_content` contains html with text outside html, example `<p></p>hello` (there is comment)


@buxx Done

File: backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py [hunk 1]
```diff
@@ -28,41 +37,53 @@ class HtmlSanitizer(object):
...
                 if elem in tag.attrs['class']:
                     return True
         if 'id' in tag.attrs:
-            for elem in HtmlSanitizerConfig.Id_blacklist:
+            for elem in self.config.id_blacklist:
                 if elem in tag.attrs['id']:
                     return True
         return False
+
+    def html_is_empty(self, soup: Tag = None) -> bool:
```
Comment (buxx) [inline backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py]: Rename `soup` into something like `node_to_inspect`

Comment (buxx) [inline backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py]: docstring missing

File: backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py [hunk 2]
```diff
@@ -28,41 +37,53 @@ class HtmlSanitizer(object):
...
+        if not soup:
+            soup = self.soup
+        if not self._is_content_empty(self.soup):
+            return False
+        return all((self.html_is_empty(child) for child in soup.children))
+
+    def is_html(self):
```
Comment (buxx) [inline backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py]: docstring missing and return type missing

File: backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py (hunk 1 above)
Comment (basilelegal) [reply backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py]: done

Comment (basilelegal) [reply backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py]: done

File: backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py (hunk 2 above)
Comment (basilelegal) [reply backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py]: done

**Comment by basilelegal** [comment]
@buxx I made all the requested changes, ready to be merged.

**Pull request merged**